.tekton/acs-konflux-tasks-push.yaml

```diff
@@ -476,12 +477,33 @@ spec:
 
             echo -n "${floating_tag}" | tee "$(results.FLOATING_TAG.path)"
 
+    - name: generate-unique-tag
+      description: Generates a unique tag to apply to the resulting images.
+        Both $(params.output-image-tag) and $(tasks.get-floating-tag.results.FLOATING_TAG) are practically floating tags
+        and will be moved when the pipeline gets re-triggered for the same commit or runs for subsequent commits
+        (respectively). Once they both get moved and there's no more tag pointing at this image, Quay will
+        garbage collect it. This creates a problem for images built in the main branch if they happen to get used in any
+        of the repos - builds in these repos will fail once the image is gone. Therefore, this task generates a unique
+        image tag for each build which won't get moved and so Quay will keep the image. There's ROX-27836 to prevent
+        garbage piling up.
+      taskSpec:
+        results:
+        - name: UNIQUE_TAG
+          description: Unique tag to apply to the resulting images.
+        steps:
+        - name: generate-unique-tag
+          image: registry.redhat.io/ubi9-minimal:latest@sha256:92b1d5747a93608b6adb64dfd54515c3c5a360802db4706765ff3d8470df6290
+          script: |
+            #!/usr/bin/env bash
+            set -euo pipefail
+            echo -n "$(params.output-image-tag)-should-persist-$( date +%Y-%m-%d-%H-%M-%S.%N )" | tee "$(results.UNIQUE_TAG.path)"
```

Comment:
I tried https://github.com/konflux-ci/build-definitions/tree/main/task/generate-labels/0.1 instead of the custom bash and it worked but the only meaningful thing there is `$ACTUAL_DATE_EPOCH` which doesn't look as nice and so I ended up doing bash for flexibility since we already have it in `get-floating-tag`.

Reply:
Do we need to keep "should-persist" string in the image tag? The uniqueness comes from the timestamp, right?

Reply:
> The uniqueness comes from the timestamp, right?

Right.

> Do we need to keep "should-persist" string in the image tag?

That was my idea to include something easily searchable in code to the tag so that when looking at this tags in Quay, we could find the code responsible for them.
I'm not attached to this and can rename or remove. WDYT?

Reply:
How about `keep-image-$(params.output-image-tag)-$( date +%Y-%m-%d-%H-%M-%S.%N )`?